### PR TITLE
User Authentication

### PR DESCRIPTION
**Pull request by saladenicoise**
Added login, signup, user and admin account creation as well as basic user authentication using sessions.

Update DB script to have already created accounts (credentials for the accounts will be sent in the discord)

### CODE REVIEW

File: db/school.sql [hunk 1]
```diff
@@ -75,15 +81,44 @@ CREATE TABLE `registrar` (
...
+INSERT INTO `student` (`id`, `first_name`, `last_name`, `address`, `email`, `phone_number`, `date_of_birth`) VALUES
+(98765432, 'Test', 'Student', 'Test Student Street', 'happygero7@gmail.com', '222-333-4444', '1648-08-14');
+
+-- --------------------------------------------------------
+
+--
+-- Table structure for table `useraccounts`
+--
+
+CREATE TABLE `useraccounts` (
```
Comment (smarkandu) [inline db/school.sql:106]: What's usually the standard for table names with more than one words?  An underscore in between maybe?  Also you could just make it a table name of one word (ie. user).

Comment (saladenicoise) [reply db/school.sql:106]: that is true, but that would require refactoring the files and I don't feel like it :) especially considering that it is a pretty minor change

**Review by smarkandu — CHANGES_REQUESTED** [review]
Changes made to school.sql break the import to MySQL.  Please correct.

**Review by smarkandu — CHANGES_REQUESTED** [review]
Getting an error about a missing file:

image

File: db/school.sql (hunk 1 above)
Comment (RemineralizedWater) [reply db/school.sql:108]: Alright, minor let's merge this

**Pull request merged**